server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java

```diff
@@ -104,251 +130,274 @@ public SeriesReader(Path seriesPath, TSDataType dataType, QueryContext context,
 
 
   public boolean hasNextChunk() throws IOException {
-    if (hasCachedFirstChunkMetadata) {
-      return true;
```

Comment:
It seems that if I keep calling hasNextChunk(), the firstChunkMetaData will move to next all the time.
you can add an if statement.
if (firstChunkMetaData != null)
    return true;

Reply:
fixed